src/msg/msg_types.h

```diff
@@ -161,6 +164,14 @@ static inline void encode(const sockaddr_storage& a, ceph::buffer::list& bl) {
 				  (unsigned char*)(&ss + 1) - dst);
   ::memcpy(dst, src, copy_size);
   encode(ss, bl);
+#elif defined(_WIN32)
+  ceph_sockaddr_storage ss{};
+  ::memcpy(&ss, &a, std::min(sizeof(ss), sizeof(a)));
+  // The Windows AF_INET6 definition doesn't match the Linux one.
```

Comment:
I guess this is a holdover from MS's behavior in the bad old 90s, when they were trying to be incompatible? Ugh. Too late to change it now though, I'm sure...

Reply:
> As a general note, this function could use some other cleanup. The __linux__ case should probably just call encode instead of open-coding it there. Some of the code in these cases could be collapsed down IMO.

I'd avoid changing unrelated code though, making this bug fix easier to merge and backport, if that's ok with you.

Reply:
I'd be in favour of cleaning up this, but have to agree with @petrutlucian94.
And only do that as a separate PR.

Reply:
@petrutlucian94 
Perhaps you can transform the `#if...#endif` into a inline function or a macro

Reply:
> Perhaps you can transform the #if...#endif into a inline function or a macro

That would probably mean rewriting the other platform specific bits as well and I'd avoid doing it as part of a bug fix.

Reply:
I guess I was a bit to cryptic in what to put in a function/macro.
I ment:
```
  if (a.ss_family == AF_INET6) {
    ss.ss_family = AF_INET6_LINUX;
  }
```
But maybe that would go into some cleanup as well, so perhaps just merge this first.

Reply:
As per Jeff's IRC message, I'm marking this as resolved:

```
#ceph-devel/2021-12-07.log:[16:10:03] <lpetrut> jlayton: hi, thanks for taking a look at the windows ipv6 fix: https://github.com/ceph/ceph/pull/43957. as mentioned in the comments, I'd prefer avoiding cleaning up unrelated code and do that as a 
separate PR, making the bug fix easier to merge/backport. would you be ok with merging it as-is?
#ceph-devel/2021-12-07.log:[16:28:29] <jlayton> lpetrut: yes, sorry. I didn't mean that you should do it in the context of 
that PR (unless you want to)
```